build.assets/tooling/cmd/difftest/git.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"bytes"
+	"os/exec"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+const (
+	// git error when file does not exists in revision
+	gitFileNotExistsError = "exists on disk, but not in"
+)
+
+// gitIsAvailable returns status of git
+func gitIsAvailable() error {
+	_, err := exec.LookPath("git")
+	return err
+}
+
+// git runs git and returns output (stdout/stderr, depends on the cmd result) and error
+func git(path string, ss ...string) (string, error) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	cmd := exec.Command("git", ss...)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	cmd.Dir = path
+
+	err := cmd.Run()
+
+	if err != nil {
+		return strings.TrimSpace(stderr.String()), trace.Wrap(err)
```

Comment:
Maybe that is just me, but for me, this is not intuitive. Error is returned as a string, and `err` can be still `nil` in some cases? You should also not use `string` if there `err != nil`. I'd always append the `stderr` to the `err` and only return `stdout` as a string if there is no error.